### PR TITLE
Aliases deletion

### PR DESCRIPTION
**Pull request by frlinw**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Remove some aliases
https://github.com/sequelize/sequelize/issues/9372

## List of aliases
| Remove | Keep | |
|---|---|---|
| insertOrUpdate | upsert | ✔️|
| find | findOne | ✔️|
| findAndCount | findAndCountAll | ✔️|
| findOrInitialize | findOrBuild | ✔️|
| beforeConnection | beforeConnect | ✔️
| [after,before]BulkDelete | [after,before]BulkDestroy | ✔️
| [after,before]Delete | [after,before]Destroy | ✔️
| hook | addHook | ✔️|
| updateAttributes | update | ✔️|
| asIs | literal | ✔️|
| Sequelize.condition | Sequelize.where | ✔️|
| Sequelize.prototype.Utils | Sequelize.Utils | ✔️|
| Sequelize.prototype.Promise | Sequelize.Promise | ✔️|
| Sequelize.prototype.TableHints | Sequelize.TableHints | ✔️|
| Sequelize.prototype.Op | Sequelize.Op | ✔️|
| Sequelize.prototype.Transaction | Sequelize.Transaction | ✔️|
| Sequelize.prototype.Model | Sequelize.Model | ✔️|
| Sequelize.prototype.Deferrable | Sequelize.Deferrable | ✔️|
| Sequelize.prototype.Error| Sequelize.Error | ✔️|
| Sequelize.prototype[error] | Sequelize[error] | ✔️|

## Particular case
| Remove | New| |
|---|---|---|
| findById, findByPrimary | findByPk | ✔️|

## DataTypes
I tried to make DOUBLE PRECISION change but this caused tests failure and I was unable to fix them. So I left DOUBLE PRECISION & NUMERIC aliases unchanged.

| Remove | Keep| |
|---|---|---|
| NONE | VIRTUAL | ✔️|
| DOUBLE PRECISION | DOUBLE  | ❌|
| NUMERIC | DECIMAL | ❌|

### CODE REVIEW

**Comment by frlinw** [comment]
Tests fixed. Don't know why codecov failed.

File: lib/dialects/mysql/connection-manager.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
```
Comment (sushantdhiman) [inline lib/dialects/mysql/connection-manager.js]: Only case for `SequelizeErrors` is changed, remove changes from this file

**Comment by sushantdhiman** [comment]
Code coverage can be ignored for this PR

File: lib/dialects/mysql/connection-manager.js (hunk 1 above)
Comment (frlinw) [reply lib/dialects/mysql/connection-manager.js]: It was for consistancy, it's the only file where sequelizeErrors is used with an upper first letter. I'm reverting.

Comment (sushantdhiman) [reply lib/dialects/mysql/connection-manager.js]: Leave it

**Comment by sushantdhiman** [comment]
Thanks @frlinw

**Comment by SimonSchick** [comment]
TY, this should make future development much less confusing!

**Comment by SimonSchick** [comment]
@sushantdhiman @frlinw you missed `hasHooks` vs. `hasHook` 🙂

**Comment by frlinw** [comment]
@SimonSchick it was a request from sushantdhiman https://github.com/sequelize/sequelize/issues/9372#issuecomment-421389263

**Pull request merged**